src/backend/distributed/commands/role.c

```diff
@@ -895,6 +906,57 @@ GenerateGrantRoleStmtsOfRole(Oid roleid)
 }
 
 
+/*
+ * GenerateSecLabelOnRoleStmts generates the SecLabelStmts for the role
+ * whose oid is roleid.
+ */
+static List *
+GenerateSecLabelOnRoleStmts(Oid roleid, char *rolename)
+{
+	ScanKeyData skey[1];
+	HeapTuple tuple = NULL;
+	List *secLabelStmts = NIL;
+
+	/*
+	 * Note that roles are shared database objects, therefore their
+	 * security labels are stored in pg_shseclabel instead of pg_seclabel.
+	 */
+	Relation pg_shseclabel = table_open(SharedSecLabelRelationId, AccessShareLock);
+	ScanKeyInit(&skey[0], Anum_pg_shseclabel_objoid, BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(roleid));
+	SysScanDesc scan = systable_beginscan(pg_shseclabel, SharedSecLabelObjectIndexId,
+										  true, NULL, 1, &skey[0]);
+
+	while (HeapTupleIsValid(tuple = systable_getnext(scan)))
```

Comment:
let's keep variables close to the places where they're used:

```suggestion
		HeapTuple tuple = NULL;
		while (HeapTupleIsValid(tuple = systable_getnext(scan)))
```